server/src/main/java/org/elasticsearch/cluster/routing/allocation/IndexMetadataUpdater.java

```diff
@@ -169,8 +169,10 @@ private static IndexMetadata updateInSyncAllocations(
                 final String allocationId;
                 if (recoverySource == RecoverySource.ExistingStoreRecoverySource.FORCE_STALE_PRIMARY_INSTANCE) {
                     allocationId = RecoverySource.ExistingStoreRecoverySource.FORCED_ALLOCATION_ID;
-                    updatedIndexMetadata = updatedIndexMetadata.withTimestampRange(
-                        updatedIndexMetadata.getTimestampRange().removeShard(shardId.id(), oldIndexMetadata.getNumberOfShards())
+                    updatedIndexMetadata = updatedIndexMetadata.withTimestampRanges(
+                        /// MP TODO: Is it safe/OK to call removeShard here twice? Not sure why we call removeShard on this to start with.
```

Comment:
Yes, I think this is needed. We're removing the shard from the two different ranges because we're creating it again from a stale copy that might in principle have different data, which will be filled in again on the subsequent shard-started event.